### PR TITLE
Add ability to specify application metadata

### PR DESCRIPTION
**Pull request by Qubad786**
## Context

For most cases we would want app developers to be limited to what they try to consume from the host but in some cases it is necessary for the app to function.
A good example is `netdata` application, in order for it to gather statistics it needs access to sysfs which is fine but an end user should be aware of what the app in question is trying to consume. Changes have been made which expose the following information for now and we can extend this as required:
```
runAsContext:
  - userName: root
     groupName: root
     gid: 0
     uid: 0
     description: "Why this needs to be done"
capabilities:
  - name: SYS_PTRACE
    description: "This is being used to do X thing"
hostMounts:
  - hostPath: "/proc"
    description: "Required by netdata for xyz"
```

### CODE REVIEW

**Review by Ornias1993 — CHANGES_REQUESTED** [review]
Not a fan.

- These are per-container settings within kubernetes. Here they are exposed as "per app", this is not making things clearer and will lead to confusion
- It's absolutely not clear (no documentation at all) what creators should and should-not be adding here (only root, all users, what else)
- There is no reason for users to trust this
- This information is, mostly, already available in the helm-chart
- It's unmaintainable, because of above issue of being per app. Simply put: each container has it's own reasons for being the way it is.

This needs back to the drawing board for the structure.
Not workable in this state.

From a TrueCharts perspective, we already have this info and build a helm chart and use it to build kubernetes objects.
Why do we need to ALSO render precisely the same information we just gave you access to, to also render this... Just to add a description?!

**Comment by Ornias1993** [comment]
Small addition:
From a security perspective, eventually the kube objects need to be scanned anyway to ensure the catalog creator doesn't lie.

Having to cross-reference this complex format, with the results of said scans needlessly complicates things and opens everyone up to false positives and negative.

So if you need to scan it anyway, limit the metadata to something like:
```
rootWarningDescription: "We need this for x"
privilegedDescription: "We need this for Y"
capabilityWarningDescription: "We need this for X"
mountsWarningDescription: "We need this for x"
```

(also see above, the privileged option is also missing from the warnings list).
KISS.

**Comment by Ornias1993** [comment]
An interesting note by Stavros, on top of the above...

Even when we just set the description, we don't KNOW what the reason is for the security issue on the time of creation of the App. Root could just-as-well be manually set by the users. But still a warning need to be in-place.

**Comment by sonicaj** [comment]
We know that we cannot validate if the app creator is being completely honest or not and we think at this point in time that it is fine. Eventually if some app creator is doing something funny and not being honest about it, it will become common knowledge at some point. So from that perspective, it is fine.

**Comment by sonicaj** [comment]
We will potentially consider parsing helm template output at some point but if we talk about actual k8s resources or converting them to k8s resources in python etc, we don't have plans to do that any time soon at least.
This file is not a requirement right now and the idea is only to enhance user experience as much as we can keeping in line with the work involved.

**Comment by Ornias1993** [comment]
> We know that we cannot validate if the app creator is being completely honest or not

But you can, you decide(!) not to... There is a distinct difference.

But even so, the points made above still stand:
- Some of those settings are NOT per-app but differ with each of the many containers and pods an App can have (having them per-app here, create a mess)
- SecurityContext even has two places they are used (container and pod)
- We don't know the security settings at App creation. For example: privileged gets turned on if devices are added (for TrueCharts, but also new-ix-common), but we don't know if that's the case on creation.

Simply put:
The location in the pipeline (file created on app creation), does not match the layer which you want to validate (settings configured AFTER app creation) you want to do.

**Review by sonicaj — APPROVED** [review]
Let's sync with UI team as well so we can give feedback to the user if this information is available
